tests/treq.py

```diff
@@ -78,8 +78,11 @@ def send_lines(self):
             yield lines
 
     def send_bytes(self):
-        for d in str(self.data.decode("latin1")):
-            yield bytes(d.encode("latin1"))
+        for d in self.data:
+            if six.PY3:
+                yield bytes([d])
```

Comment:
I don't think the cast is necessary anymore, since we open the file in binary mode.

Reply:
I'm not sure what was the intention there, but it's reading the data byte-by-byte and yielding byte-by-byte. 

In py2, iterating the data str will yield individual bytes as str
In py3, iterating the data bytes will yield individual bytes as int, which then need to be returned as new bytes using bytes([d])

Previous code was just plain wrong under py2 as str(self.data.decode("latin1")) will try to re-encode an unicode string in ascii

Reply:
Thanks for the explanation.